models/intermediate/int_zendesk__schedule_spine.sql

```diff
@@ -1,364 +1,256 @@
-), gap_starter as (
-    select 
-        *,
-        max(period_end) over (partition by schedule_id) as max_valid_until,
-        last_value(next_start) over (partition by schedule_id, period_start order by valid_until rows between unbounded preceding and unbounded following) as lead_next_start,
-        first_value(prev_end) over (partition by schedule_id, valid_from order by start_time_utc rows between unbounded preceding and unbounded following) as first_prev_end
-    from non_holiday_period_adjustments
-
--- There may be gaps in holiday and non holiday schedules, so we need to identify where these gaps are
-), gap_adjustments as(
+        schedule_name,
+        valid_from,
+        valid_until,
+        holiday_name,
+        holiday_valid_from,
+        holiday_valid_until,
+        holiday_starting_sunday,
+        holiday_ending_sunday,
+        holiday_start_or_end,
+        valid_from_index,
+        case when holiday_start_or_end = '1_holiday'
+            then 'holiday'
+            else change_type
+            end as change_type
+    from adjust_ranges
+    -- filter out irrelevant records after adjusting the ranges
+    where not (valid_from >= valid_until and holiday_date is not null)
+
+-- Converts holiday valid_from and valid_until times into minutes from the start of the week, adjusting for timezones.
+), valid_minutes as(
+    select
+        holiday_weeks.*,
 
-    select 
-        *,
-        -- In order to identify the gaps we check to see if the valid_from and previous valid_until are right next to one. If we add two hours to the previous valid_until it should always be greater than the current valid_from.
-        -- However, if the valid_from is greater instead then we can identify that this period has a gap that needs to be filled.
-        case 
-        when cast({{ dbt.dateadd("hour", "2", "valid_until") }} as {{ dbt.type_timestamp() }}) < cast(lead_next_start as {{ dbt.type_timestamp() }})
-            then 'gap'
-        when (lead_next_start is null and valid_from < max_valid_until and period_end != max_valid_until)
-            then 'gap'
+        -- Calculate holiday_valid_from in minutes from week start
+        case when change_type = 'holiday' 
+            then ({{ dbt.datediff('holiday_starting_sunday', 'holiday_valid_from', 'minute') }}
+                - offset_minutes) -- timezone adjustment
             else null
-        end as is_schedule_gap
-
-    from gap_starter
+        end as holiday_valid_from_minutes_from_week_start,
 
--- We know where the gaps are, so now lets prime the data to fill those gaps
-), schedule_spine_primer as (
+        -- Calculate holiday_valid_until in minutes from week start
+        case when change_type = 'holiday' 
+            then ({{ dbt.datediff('holiday_starting_sunday', 'holiday_valid_until', 'minute') }}
+                + 24 * 60 -- add 1 day to set the upper bound of the holiday
+                - offset_minutes)-- timezone adjustment
```

Comment:
```suggestion
                - offset_minutes) -- timezone adjustment
```
I feel like I've seen this cause issues before

Reply:
Ahh. Updated!